runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/operation/HttpOperationContext.kt

```diff
@@ -80,16 +77,18 @@ public open class HttpOperationContext {
          * (Optional) prefix to prepend to a (resolved) hostname
          */
         public var hostPrefix: String? by option(HostPrefix)
+
+        /**
+         * The top-level trace span for this request. This span will be used to generate child spans for inner parts of
+         * the request.
+         */
+        public var traceSpan: TraceSpan? by requiredOption(TracingContext.TraceSpan)
```

Comment:
Any reason to not default this to `NoOpTraceSpan`?

Reply:
Yes, any place that uses `NoOpTraceSpan` will effectively drop tracing events irrespective of whatever probe the user has configured. That seemed like a dangerous default to apply when the risk would be _not_ logging things. I chose to make the parameter required and provide no default in order to find/fix all the places where trace spans would be used but where none was wired up.

Adding a default of `NoOpTraceSpan` would simplify various unit tests but would introduce the possibility of silently dropping trace events in new code that wants a trace span.

Reply:
Yes and no. In generated code we know we are going to wire up a real span so it is mostly unit tests that bear the brunt of this. I don't have a strong opinion on this but I've definitely had moments where I've wondered if some of our other required options should truly be required (e.g. setting the `service` didn't make much sense when using our HTTP abstractions to implement various credential providers)